students/etwum/lesson05/mailroom_part3.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+
+
+# initial list of donors
+lst_donors = [{"donor name": "Lionel Messi", "total donations": 1000000.00, "number donations": 5, "avg donation": 20000.00},
+              {"donor name": "Thierry Henry", "total donations": 500, "number donations":1, "avg donation": 500},
+              {"donor name": "Michael Jordan", "total donations": 45000, "number donations": 3, "avg donation": 15000},
+              {"donor name": "Kobe Bryant", "total donations": 8000, "number donations": 2, "avg donation": 4000}]
+
+
+def send_thank_you():
+    # used to add new donors
+    new_donor = []
+
+    # function creates a thank you email to current and new donors added to the list
+    # add new donors and donations
+    # print the names of the current donor if 'list' is input by the user
+    print('\n-----------------------------------------------------')
+    view_donors = input("If you would like to see a list of donors please type 'list' or any key to continue. ")
+    if view_donors == 'list':
+        print()
+        print("Below are the list of donors:")
+        for x in lst_donors:
+            print(x["donor name"])
+        print('----------------------------')
+
+    while True:
+
+        # validates to make sure the user inputs a first and last name
+        try:
+            # input donor name to print an email
+            donor_first_name = input("Please input the first name of the donor. ")
+            donor_last_name = input("Please input the last name of the donor. ")
+            if donor_first_name == "" or donor_last_name == "":
+                raise Exception
+        except Exception:
+            print("\nPlease input a first and last name")
+            print()
+        else:
+            break
+
+    donor_full_name = donor_first_name.capitalize() + " " + donor_last_name.capitalize()
+    print('-----------------------------------------------------')
+    print()
+
+    # prints an email if the donors name is currently in the list
+    for x in lst_donors:
+        if x["donor name"] == donor_full_name:
+
+            while True:
+
+                # raise exception if the user inputs anything other than 'yes' or 'no'
+                try:
+                    str_donation_choice = input("Would you like to add another donation? (yes or no) ")
+                    if str_donation_choice not in ("yes","no"):
+                        raise Exception
```

Comment:
adding the try except here seems a little extra, could have been done simply with if/else which is already there